src/transaction/log_replication_atomic.cpp

```diff
@@ -123,7 +124,7 @@ namespace cublog
 		m_replicator_mvccid->complete_mvcc (header.trid, replicator_mvcc::COMMITTED);
 	      }
 
-	    if (m_locked_classes.count (header.trid) > 0)
+	    if (m_locked_objects.count (header.trid) > 0)
```

Comment:
Does a change to the _db_serial trigger to invalidate all those caches?

Reply:
For readability, I suggest two options:
1. `m_locked_objects` -> `m_locked_objects_for_ddl`. It's simple, but honestly I don't like the big `if` in `discard_caches_for_ddl`. But, OK.
2. Separating the map for each db_serial and Rootclass, and do what it's needed to do explicitly.
```
if (m_locked_class.count () > 0)
{
  discard_caches_for_ddl (thread_entry, header.trid);
}
if (m_locked_class.count () + m_locked_serial.count() > 0)
{
 release_all_locks_for_ddl (thread_entry, header.trid);
}
```
3. or add a comment.

Reply:
Separating the map will be clear, I will do as second one.